### DISCUSSION TITLE
Adding new lint rules to an existing codebase (Continuous Accessibility Panel)

### DISCUSSION THREAD

**Discussion by Wynston** [community/events#55] [Community]
In Melanie's presentation about Continuous Accessibility, they talk about a strategy for adding new lint rules to an existing codebase. 

Rather than trying to fix all the issues before turning the rule on, they suggest enabling the rule and using **dated TODO comments** that will eventually begin to throw lint warnings, and then lint errors after X amount of time.

Does anyone know how this can be accomplished?

For example, if I wanted to add the [eslint accessibility plugin for JSX](https://www.npmjs.com/package/eslint-plugin-jsx-a11y), is there a way to generate dated TODO comments in a way that eslint could support, which would eventually provide warnings and errors?

**Comment by ossanna16** [comment]
Hi there @Wynston and thank you for the great question! Is [this](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md) perhaps helpful to you?

**Reply by Wynston** [reply]
@ossanna16 thank you!

I think that would work. However, it seems like it would be a very tedious task to add all those todos manually. 

The ideal process would be:
1. Add a new lint rule 
2. Autogenerate dated todo comments for lines that have errors related to that new rule
  a. the todo comment would disable the lint error until the date has passed
  
It would be a bit of a ticking time bomb but it would be easy enough to push back the date if necessary. The primary benefit is to protect new code and this process would remove the friction of a massive initial refactor.

**Reply by IanEdington** [reply]
I've used this approach in the past and it worked really well. A huge bonus is that in code reviews if you see one of these comments around the code you can ask for it to be removed. I much prefer this approach to the one where there's a list of ignored errors in a file no one looks at.

In the past, I've used the custom output formatting on eslint to generate a long list of search and replace commands using sed.
I ran the linter and dumped the output to a file, then added a `#! /bin/bash` and ran the file.
The files were then auto-formatted so the comments were in line with the code and it was good to go.

I wish I'd open sourced that code, I'm actually looking to do the same thing again and wanted to see if anyone had written such a thing.

**Comment by xTalhaAhmedCho** [comment]
![chrome_eTjRY3DcEQ](https://github.com/user-attachments/assets/247865c8-f382-48c5-a43a-de6bd50d3803)

`Dear GitHub Support Team,`

> I am writing to request the re-enablement of GitHub Actions for my account, **@TalhaAhmedCho**. I have noticed that GitHub Actions is currently disabled for my account. 

I understand that my GitHub Actions may have been disabled due to `misunderstanding of limits`. If my actions inadvertently violated any GitHub terms or policies, I sincerely apologize. It was not my intention to misuse the platform. I am committed to using GitHub Actions responsibly and in accordance with all guidelines in the future.

> I value GitHub's services greatly, especially GitHub Actions, which is an **_integral part of my workflow_** for personal project. I rely on GitHub Actions for `learning new technologies. `

**_I kindly request you to review my account and re-enable GitHub Actions. I am willing to cooperate fully and take any necessary steps to ensure I comply with GitHub's policies and guidelines in the future._**

`My GitHub username is:`  **@TalhaAhmedCho**

Thank you for your **_time and consideration_**. I look forward to your positive response and the re-enablement of GitHub Actions.

Sincerely,

Talha Ahmed Chowdhury
[Linkedin](https://www.linkedin.com/in/TalhaAhmedCho/) | [Facebook](https://www.facebook.com/@TalhaAhmedCho) | [YouTube](https://www.youtube.com/@TalhaAhmedCho) | [Twitter](https://x.com/_TalhaAhmedCho)
